### PR TITLE
Rework USB Audio Class stuff

### PR DESCRIPTION
**Pull request by twam**
Replaces #18 and #19.

I was able to implement an UAC2 audio device with input and output with these changes.

### CODE REVIEW

File: usb_protocol/types/descriptors/uac3.py [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
 #
 """
-    descriptors specific to USB version 2 
+    Descriptors for USB Audio Class Devices (UAC), Release 3
+
+    [Audio30] refers to "Universal Serial Bus Device Class Definition for Audio Devices", Release 3.0, September 22, 2016
+    [Frmts30] refers to "Universal Serial Bus Device Class Definition for Audio Data Formats", Release 3.0, September 22, 2016
+    [TermT30] refers to "Universal Serial Bus Device Class Definition for Terminal Types", Release 3.0, May 31, 2006
+
     NOTE: This is not complete yet and will be extended as needed
```
Comment (hansfbaier) [inline usb_protocol/types/descriptors/uac3.py:11]: Is this comment still valid? It looks like you really fleshed the implementation out a bit. What would be missing?

**Review by hansfbaier — APPROVED** [review]
Thanks for the work you put into it.
I tried to factor out the commonalities in my first attempt but
I agree that it is so much tidier this way.
Great work!

File: usb_protocol/types/descriptors/uac2.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
 #
 # This file is part of usb-protocol.
 #
-""" descriptors specific to USB version 2
+"""
+    Descriptors for USB Audio Class Devices (UAC), Release 1
```
[the remark this answers is not in the document]
Comment (twam) [reply usb_protocol/types/descriptors/uac2.py:5]: Yes!

File: usb_protocol/types/descriptors/uac3.py (hunk 1 above)
Comment (twam) [reply usb_protocol/types/descriptors/uac3.py:11]: Yes, there are still a lot of descriptors missing, e.g. like the FeatureUnit.

Comment (hansfbaier) [reply usb_protocol/types/descriptors/uac3.py:11]: OK. Thanks!

**Comment by twam** [comment]
> Thanks for the work you put into it.
> I tried to factor out the commonalities in my first attempt but
> I agree that it is so much tidier this way.
> Great work!

Thanks for the appreciation.
Some of the differences are really subtle, so I thought this way it is much clearer to focus on the actual spec. version you're using.

**Comment by twam** [comment]
Fixed the typo found by @hansfbaier and another typo in the commit message. Removed an accidentally committed file. :)

**Comment by ktemkin** [comment]
Thanks a ton! ^_^

**Pull request merged**